Kernel/Term.cpp

```diff
@@ -332,17 +332,22 @@ bool Term::isApplication() const {
   return !isSort() && !isLiteral() && env.signature->isAppFun(_functor);    
 }
 
+unsigned Term::numTypeArguments() const {
+  CALL("Term::numTypeArguments");
+
```

Comment:
Good idea. Btw - I lifted this implementation from `Term::termArgs` directly below, which includes the `isSpecial` check. However, the implementation from `Term::hasTermArgs` doesn't. Any potential problem here?

Reply:
I can see no reason for the absence of the `isSuper()` check in `hasTermArgs`. I am about to push a fix that replaces the function `hasTermArgs` with one called `numTermArguments`. It can obviously be used in a context requiring a Boolean as well, but has nicer symmetry  with `numTypeArguments`.